src/rewards/airdrop.ts

```diff
@@ -205,15 +205,12 @@ export class Airdrop {
     return new Airdrop(airdropAddress, rewardsVaultAddress, airdrop, rewardsVault)
   }
 
-  static async loadAll(rewardsProgram: Program, stakePoolVault: PublicKey): Promise<Airdrop[]> {
-    const stakePoolFilter: MemcmpFilter = {
-      memcmp: {
-        offset: 8,
-        bytes: stakePoolVault.toBase58()
-      }
+  static async loadAll(rewardsProgram: Program): Promise<Airdrop[]> {
+    const dataSizeFilter: DataSizeFilter = {
+      dataSize: 400464
     }
+    const airdropInfos = await rewardsProgram.account.airdrop.all([dataSizeFilter])
```

Comment:
Filtering is generally good
The datasize filter is welcome, it filters out all the old corrupted airdrops
I think you want the stake vault filter as well, we don't care about other stakepools

```suggestion
  static async loadAll(rewardsProgram: Program, stakePoolVault: PublicKey): Promise<Airdrop[]> {
    const dataSizeFilter: DataSizeFilter = {
      dataSize: 400464
    }
    const stakePoolFilter: MemcmpFilter = {
      memcmp: {
        offset: 8,
        bytes: stakePoolVault.toBase58()
      }
    }
    const airdropInfos = await rewardsProgram.account.airdrop.all([dataSizeFilter, stakePoolFilter])
```

Reply:
good idea =]